tools/sof-dump-status.py

```diff
@@ -87,10 +87,20 @@ def loadPCI(self):
                     break
             pci_info['hw_id']="0x" + tmp_line[2] + tmp_line[1] + " 0x" + tmp_line[4] + tmp_line[3]
             pci_info['hw_name'] = self._pci_ids["0x" + tmp_line[4] + tmp_line[3]]
-            if pci_info['hw_name'] in apci_key_lst:
-                pci_info['device_id'] = self._acpi_ids[pci_info['hw_name']]
             self.pci_lst.append(pci_info)
 
+    def loadACPI(self):
+        self.acpi_lst.clear()
+
+        for mach in self._acpi_ids.keys():
+            acpi_info = {}
+            exit_code, output=subprocess.getstatusoutput("cat /sys/bus/acpi/devices/%s:00/status" % (self._acpi_ids[mach]))
+            if exit_code != 0:
+                continue
+            acpi_info['hw_name'] = mach
+            acpi_info['acpi_id'] = self._acpi_ids[mach]
+            self.acpi_lst.append(acpi_info)
```

Comment:
Shouldnt we just break when one of the ID's is found?

Reply:
Don't ask me why a list is used. This is the same code we've been using *for months*, I just solved the ACPI/PCI confusion.

Whoever wrote that script made a lot of assumptions that made no sense, I just found that the DMI parsing is incorrect as well.

Reply:
> Whoever wrote that script made a lot of assumptions that made no sense, I just found that the DMI parsing is incorrect as well.

Please at least add `FIXME: why is this a list?` comments so your time reverse engineering this is not completely wasted.

Reply:
I'll add a separate bug, there are lists for PCI everywhere.